### PR TITLE
Add missing documentation

### PR DESCRIPTION
**Pull request by larsk21**
JavaDoc and comments for:
- GermanState enum
- everything in main package

(the indentation fixes messed up the diff, I'm sorry)

Please check if I've forgotten anything.
Also, if you need any more documentation, just tell me ;)

### CODE REVIEW

File: src/main/java/main/UserInput.java [hunk 1]
```diff
@@ -8,170 +8,207 @@
...
+                    if (branch.startsWith("$")) {
+                        branch = null;
+                    }
+                    commit = buildProperties[3];
+                    if (commit.startsWith("$")) {
+                        commit = null;
+                    }
+                }
+            }
+        } catch (IOException e) {}
```
Comment (gstuer) [inline src/main/java/main/UserInput.java]: Oh, I think we forgot to fill this catch block here? Not part of this PR, nevertheless, fixing that would be great!

File: src/main/java/main/UserInput.java [hunk 2]
```diff
@@ -8,170 +8,207 @@
...
+     * @param args Command line arguments that the application was started with
+     */
+    public UserInput(String[] args) {
+        this.args = args;
+    }
+
+    /**
+     * Parse the command line arguments
+     * @return The user request read from the command line arguments
+     * @throws ParseException Thrown in caes the command line arguments are invalid
```
Comment (gstuer) [inline src/main/java/main/UserInput.java]: Just a little typo here => "Thrown in **_caes_** the..."

File: src/main/java/main/UserInput.java [hunk 3]
```diff
@@ -8,170 +8,207 @@
...
+            case JSON_GLOBAL:
+                file = new File(fileArgs[0]);
+                break;
+            case JSON_MONTH:
+                file = new File(fileArgs[1]);
+                break;
+            case OUTPUT:
+                file = new File(fileArgs[2]);
+                break;
+            default: // never userd
```
Comment (gstuer) [inline src/main/java/main/UserInput.java]: Little Typo :)

File: src/main/java/main/UserInput.java (hunk 1 above)
Comment (larsk21) [reply src/main/java/main/UserInput.java]: This is left out on purpose. The relevant variables are set to null before the try-catch block, so if none of them is set, the error will be handled in the if-clause below.
